frontend/src/routes/UserManagement/Roles/Role/RolePermissions.tsx

```diff
@@ -1,18 +1,111 @@
 /* Copyright Contributors to the Open Cluster Management project */
-import { PageSection } from '@patternfly/react-core'
-import { useParams } from 'react-router-dom-v5-compat'
+import { Flex, FlexItem, Label, PageSection } from '@patternfly/react-core'
+
 import { useTranslation } from '../../../../lib/acm-i18next'
-import { AcmPage, AcmPageContent, AcmPageHeader } from '../../../../ui-components'
+import { AcmPage, AcmPageHeader, AcmTable, compareStrings, IAcmTableColumn } from '../../../../ui-components'
+
+import { PolicyRule } from '../../../../resources/kubernetes-client'
+import { useMemo, useCallback } from 'react'
+import { useCurrentRole } from '../RolesPage'
+import { cellWidth } from '@patternfly/react-table'
+
+export const kindToAbbreviation = (resourceOrKind: string): string => {
+  // Step 1: Extract the resource name part (before any path separators)
+  const resourceName = resourceOrKind.split(/[/\\?#]/)[0] || resourceOrKind
+
+  // Complex words: virtualmachines → VIRT, persistentvolumes → PERS
+  const abbrKind = resourceName.toUpperCase().slice(0, 4)
+  // Using logic in openshift console code
+  // https://github.com/openshift/console/blob/a228536605a2451468d172da9bb7b6ab812511ea/frontend/public/module/k8s/get-resources.ts#L29
+  const blocklist = ['ASS', 'FART']
+  return blocklist.includes(abbrKind) ? abbrKind.slice(0, -1) : abbrKind
+}
 
 const RolePermissions = () => {
   const { t } = useTranslation()
-  const { id = undefined } = useParams()
+  const currentRole = useCurrentRole()
+
+  const policyRules = useMemo(() => {
+    if (!currentRole?.rules) return []
+    return currentRole.rules
+  }, [currentRole])
```

Comment:
see [useMemo](https://react.dev/reference/react/useMemo)

there's no computation effort here, it is just about to get rules or [], so my recomendation is not to use `useMemo` hook. Additionally to it, `policyRules` is used on one single place, so I would remove policyRules and do

```
        <AcmTable<PolicyRule>
          id="permissions-table"
          key="permissions-table"
          columns={columns}
          keyFn={keyFn}
          items={currentRole?.rules ?? []}
          emptyState={<div>{t('No permissions found')}</div>}
          autoHidePagination={true}
          initialPerPage={100}
          fuseThreshold={0.1}
        />
```